Sources/TuistCore/Models/TargetAction.swift

```diff
@@ -13,20 +13,59 @@ public struct TargetAction: Equatable {
         case post
     }
 
+    /// How to execute the target action
+    ///
+    /// - file: Executes the tool, calling the script at the path. Tuist will look up the tool on the environment's PATH
+    /// - text: Executes the embedded script. This should be a short command.
+    private enum Script: Equatable {
+        case externalFile(_ tool: String?, _ path: AbsolutePath?, _ args: [String])
+        case embedded(String)
+    }
```

Comment:
+1 I've already made that note [here](https://github.com/tuist/tuist/pull/2192#discussion_r548036181) and @jsorge is on it 🙂

Reply:
Thanks! apologies missed that comment

Reply:
In doing this update it feels like it could be a breaking change (even though it's exposing the actual correct behavior). Currently Tuist does support sending both a `tool` and a `path`, and this change would make that not the case. My concern is making this source-breaking change. Do you all think that risk is worth taking right now @kwridan @fortmarek?

Reply:
Even though the public interface supports passing both, which in hindsight it was not a good idea, Tuist [only uses one](https://github.com/tuist/tuist/pull/2192/files#diff-76fb8ea4f5a0578f099fe12dfa280a0f19d2f272929791eb8832c7469b522c51L95). That being said, I think it's a safe breaking change to introduce. If doubt developers were passing both arguments at the same time, and if they were, they were misusing a badly-designed API.
That being said, I'd go ahead and apply @fortmarek & @kwridan 's suggestion.

Reply:
@kwridan @fortmarek @pepibumur I've [pushed a commit](https://github.com/tuist/tuist/pull/2192/commits/82fe96fb95907d7c69747e0491cd9b2f25642f0c) with this refactor, but some of the tests which are not related to my changes are now failing and I'm not sure why. Any ideas? Attached is an image of the failures I'm seeing.

![Screen Shot 2020-12-29 at 11 59 07 AM](https://user-images.githubusercontent.com/2585841/103310780-433b1180-49cd-11eb-8c8a-9a6e697f3c81.png)

Reply:
Nevermind, looks like that failure was on my machine for some reason since it passed CI. We should be good to merge!